services/thermal_mgr/thermal_mgr.c

```diff
@@ -44,17 +45,56 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
 
+  if (thermalMgrQueueHandle == NULL || event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+
+  xQueueSend(thermalMgrQueueHandle, event, (TickType_t) 0);
+
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t osEvent = {0};
+  osEvent.type = THERMAL_MGR_EVENT_OS_CMD;
+
+  thermalMgrSendEvent(&osEvent);
 }
 
-static void thermalMgr(void *pvParameters) {
+static void thermalMgr(void *pvParameters) { // what is pvParameters used for?
   /* Implement this task */
+
+  error_code_t errCode;
+  
+  thermal_mgr_event_t queueEventItemBuffer = {0};
+
   while (1) {
     
+    if (xQueueReceive(thermalMgrQueueHandle, &queueEventItemBuffer, (TickType_t) portMAX_DELAY) == pdPASS) {
+
+      if (queueEventItemBuffer.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD) {
+
+        float tempC;
+        LOG_IF_ERROR_CODE(readTempLM75BD(LM75BD_OBC_I2C_ADDR, &tempC));
+
+        addTemperatureTelemetry(tempC);
+
+      } else if (queueEventItemBuffer.type == THERMAL_MGR_EVENT_OS_CMD) {
+
+        float tempC;
+        LOG_IF_ERROR_CODE(readTempLM75BD(LM75BD_OBC_I2C_ADDR, &tempC));
+
+        if (tempC > LM75BD_DEFAULT_HYST_THRESH) {
+          overTemperatureDetected();
+        } else {
+          safeOperatingConditions();
+        }
```

Comment:
You shouldn't get to these lines if the readTemp fails. Add an `if(errCode != success) continue;` line before this